app/Http/Transformers/SignupTransformer.php

```diff
@@ -48,12 +56,31 @@ public function transform(Signup $signup)
      * Include posts
      *
      * @param \Rogue\Models\Signup $signup
+     * @param \League\Fractal\ParamBag|null
      * @return \League\Fractal\Resource\Collection
      */
-    public function includePosts(Signup $signup)
+    public function includePosts(Signup $signup, ParamBag $params = null)
     {
-        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
-        $post = $signup->visiblePosts;
+        if ($params === null) {
+            // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
+            $post = $signup->visiblePosts;
+
+            return $this->collection($post, new PostTransformer);
+        }
+
+        // Optional params validation
+        $usedParams = array_keys(iterator_to_array($params));
+
+        if ($invalidParams = array_diff($usedParams, $this->validParams)) {
+            throw new \Exception(sprintf(
```

Comment:
This should probably be a `ValidationException` so it gets formatted as a 422:

```php
use Illuminate\Validation\ValidationException;

// ...

throw ValidationException::withMessages([
    'include' => 'Parameters must be one of: ' . implode(',', $this->validParams),
]);
```

Reply:
ah great call - updated! :)